src/components/NavigationModal.js

```diff
@@ -0,0 +1,83 @@
+const React = require('react');
+const _merge = require('lodash/merge');
+
+const Icon = require('./Icon');
+
+const StyleConstants = require('../constants/Style');
+
+const NavigationModal = React.createClass({
+  propTypes: {
+    componentStyle: React.PropTypes.object,
+    iconSize: React.PropTypes.number,
+    iconStyle: React.PropTypes.object,
+    iconType: React.PropTypes.string,
```

Comment:
scratch the deleted comment.  I see how you're protecting against them.  Would it not be easier to supply default values for these rather than doing conditional for the Icon component?